### ISSUE TITLE
Stringification of response status codes leads to invalid swagger

### ISSUE THREAD

**Issue by DStape** [marshmallow-code/apispec#426]
Recent merge of https://github.com/marshmallow-code/apispec/pull/424 results in our swagger documentation being generated with the following response definitions:

```
"responses": {
          "HTTPStatus.OK": {
```

Previously:
```
"responses": {
          "200": {
```
i.e. usage of https://docs.python.org/3/library/http.html#http-status-codes results in the generation of invalid swagger.

Potential fix (```get_ref()``` in ```/src/apispec/core.py```):
```
for code, response in iteritems(operation["responses"]):
    try:
        status_code = int(code)
    except TypeError, ValueError:
        # log error and skip
     responses[str(status_code)] = get_ref("response", response, openapi_major_version)
```

I'm happy to open a PR for this, but would appreciate any feedback on the approach laid out above (cast the code to an int before casting to str).

Thanks,
David

**Comment by sloria** [comment]
I think the try-except approach makes sense. @lafrech ?

**Comment by lafrech** [comment]
I'm not a big fan of the try-except approach where we try to catch all kinds of errors, potentially hiding input errors. And I wouldn't log an error since this code raises on `'4XX'`, for instance, which is valid as per #422.

We only ever tested/supported status codes as string or integers. AFAIU, `HTTPStatus` worked because it's a subclass of `IntEnum` and it would serialize as integer when json dumped.

Since `HTTPStatus` is part of Python core and is totally relevant here, it would be nice to support it and I think we can do it explicitly:

```py
            for code, response in iteritems(operation["responses"]):
                if not PY2:
                    from http import HTTPStatus
                    if isinstance(code, HTTPStatus):
                        code = code.value
                responses[str(code)] = get_ref(
                    "response", response, openapi_major_version
                )
            operation["responses"] = responses
```

**Comment by DStape** [comment]
Will take your comments on board and will have a PR out soon.

**Comment by DStape** [comment]
PR #436

**Comment by sloria** [comment]
The fix is released in 1.3.1.
